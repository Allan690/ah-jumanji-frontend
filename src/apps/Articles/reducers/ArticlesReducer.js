import {
    POSTING_ARTICLE,
    DELETING_ARTICLE,
    POSTED_ARTICLE,
    GOT_ARTICLE,
    ERROR_GETTING_ARTICLE,
    UPDATING_ARTICLE,
    UPDATED_ARTICLE,
    UPDATING_CURRENT,
    UPDATED_ERROR,
    ERROR_DELETING_ARTICLE,
    DELETED_ARTICLE
  } from '../actions/actionTypes';


export default function articlesReducer(state = {}, action) {
    switch (action.type) {
        case POSTING_ARTICLE:
            return Object.assign({}, state, action.payload);
        case POSTED_ARTICLE:
            return Object.assign({}, state, action.payload);
        case GOT_ARTICLE:
            return Object.assign({}, state, action.payload);
        case ERROR_GETTING_ARTICLE:
            return Object.assign({}, state, action.payload);
        case UPDATING_ARTICLE:
            return Object.assign({}, state, action.payload);
        case UPDATED_ARTICLE:
            return Object.assign({}, state, action.payload);
        case UPDATING_CURRENT:
            return Object.assign({}, state, action.payload);
        case UPDATED_ERROR:
            return Object.assign({}, state, action.payload);
        case ERROR_DELETING_ARTICLE:
            return Object.assign({}, state, action.payload);
        case DELETING_ARTICLE:
            return Object.assign({}, state, action.payload);
        case DELETED_ARTICLE:
            return Object.assign({}, state, action.payload);
        default:
            return state;
    }
}