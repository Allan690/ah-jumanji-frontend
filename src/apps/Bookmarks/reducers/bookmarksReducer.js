import * as types from  '../actions/actionTypes';

export const initialState = {
    bookmark: {bookmarks: {}},
    slug: "",
    bookmarked: false
  };

export default function bookmarksReducer(state = initialState, action){
    switch (action.type) {
        case types.BOOKMARK_SUCCESS:
            return {...state, bookmarks: action.bookmarks};

        case types.BOOKMARK_FAILED:
            return Object.assign({}, state, action.bookmarkError);

        default:
            return state;
    }
}
