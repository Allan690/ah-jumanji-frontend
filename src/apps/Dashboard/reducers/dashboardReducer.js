import * as types from  '../actions/actionTypes';


export default function dashboardReducer(state = {}, action){
    switch (action.type) {
        case types.FETCH_SUCCESS:
            return {...state, articles: action.articlesData};

        case types.FETCH_FAILED:
            return Object.assign({}, state, action.error);

        default:
            return state;
    }
}
