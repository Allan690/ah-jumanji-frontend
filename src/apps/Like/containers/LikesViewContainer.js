import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import LikesViewComponent from "../components/LikesViewComponent";
import {fetchLikers} from "../actions/likeActions";


class LikesViewContainer extends Component {
  componentDidMount = () => {
    const {fetchLikers} = this.props;
    const slug = "mr-python"; // Not to be hard coded
    fetchLikers(slug);
  };

  render() {
    const {reactions} = this.props;

    return (
      <div className="tab-pane fade show" id="likes" role="tabpanel" aria-labelledby="tab">
        <div className="col-md-12">
          <ul id="likes-list" className="list-group list-group-flush likers">
            {
              reactions ? reactions.map(rxn => (
                <LikesViewComponent key={rxn.id} reaction={rxn} />
              )) : <span />
            }
            <br />
          </ul>
        </div>
      </div>
    );
  }
}

LikesViewContainer.propTypes = {
  reactions: PropTypes.array.isRequired,
  fetchLikers: PropTypes.func.isRequired,
};

const mapStateToProps = ({LikeReducer}) => LikeReducer;

export default connect(mapStateToProps, {fetchLikers})(LikesViewContainer);
