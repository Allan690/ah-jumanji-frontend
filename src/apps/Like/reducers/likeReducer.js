import * as types from '../actions/likeActionTypes';

export const initialState = {
    likeCount: 0,
    liked: false,
    reactions: []
  };

export default(state=initialState, { type, payload }) => {
  switch(type) {
    case types.LIKE_SUCCESS:
      return {
        ...state,
        liked: !state.liked,
        likeCount: state.likeCount
      };

    case types.UNLIKE_SUCCESS:
      return {
        ...state,
        liked: state.liked,
        likeCount: state.likeCount
      };

    case types.GET_LIKES:
      return {
        ...state,
        likeCount: payload.likes,
      };

    case types.FETCH_LIKERS_SUCCESS:
      return {
        ...state,
        reactions: payload
      };

    default:
      return state;
  }
};
