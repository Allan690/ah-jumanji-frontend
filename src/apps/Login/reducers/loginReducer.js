import * as types from  '../actions/actionTypes';


export default function loginUserReducer(state = {}, action){
    switch (action.type) {
        case types.LOGIN_SUCCESS:
            return Object.assign({}, state, action.user);

        case types.LOGIN_FAILED:
            return Object.assign({}, state, action.error_message);

        default:
            return state;
    }
}
