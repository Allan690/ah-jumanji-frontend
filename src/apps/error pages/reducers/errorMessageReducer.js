import * as types from '../actions/actionTypes';

export default function ErrorHandlingReducer(state = {}, action) {
  switch (action.type) {
    case types.NEW_ERROR_RAISED:
      return Object.assign({}, state, action);
    default:
      return state;
  }
}
