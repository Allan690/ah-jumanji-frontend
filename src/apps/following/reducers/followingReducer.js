import * as types from  '../actions/types';

const initialState = {
    followUser: {},
    unfollowUser: {},
    followers: {},
    following: {}
};

export default function followingReducer(state = initialState, action){
    switch (action.type) {
        case types.FOLLOW:
            return {
                ...state,
                followUser: action.user
            };
        case types.UNFOLLOW:
            return {
                ...state,
                unfollowUser: action.message
            };
        case types.GET_FOLLOWERS:
            return {
                ...state,
                followers: action.followers
            };
        case types.GET_FOLLOWED:
            return {
                ...state,
                following: action.followed
            };
        default:
            return state;
    }
}
