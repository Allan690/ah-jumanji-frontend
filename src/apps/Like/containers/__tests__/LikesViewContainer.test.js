import React from 'react';
import Enzyme, {mount, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import thunk from 'redux-thunk';
import {Provider} from 'react-redux';
import {MemoryRouter} from 'react-router-dom';
import configureStore from 'redux-mock-store';

import {initialState} from "../../reducers/likeReducer";
import LikesViewContainer from "../LikesViewContainer";

Enzyme.configure({adapter: new Adapter()});
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('<LikesViewContainer />', () => {
  it('should match the snapshot', () => {
    const component = shallow(<LikesViewContainer debug />);
    expect(component).toMatchSnapshot();
  });
});

describe('<LikeViewContainer />', () => {
  let store;
  let wrapper;

  const props = {
    likeCount: 2,
    liked: true,
    reactions: [
      {
        article: "mr-ruby",
        reaction: "1",
        user: {
          username: "Masha",
          bio: "",
          profile_pic: ""
        }
      }
    ],
    slug: 'mr-python',
    addLikes: jest.fn(),
    getArticleLikes: jest.fn(),
    fetchLikers: jest.fn(),
  };

  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = mount(
      <Provider store={store}>
        <MemoryRouter>
          <LikesViewContainer {...props} />
        </MemoryRouter>
      </Provider>
    );
  });

  it('renders without crashing', () => {
    const snapShot = shallow(<LikesViewContainer />);
    expect(snapShot).toMatchSnapshot();
  });
});
