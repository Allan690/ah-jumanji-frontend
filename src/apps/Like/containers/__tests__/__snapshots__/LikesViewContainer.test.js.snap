// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LikeViewContainer /> renders without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`<LikesViewContainer /> should match the snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
